cds/modules/stats/views.py

```diff
@@ -91,117 +91,120 @@ def _build_subquery(report_number):
+                    stat,
+                    key_type)
+
+                # Get timestamp-aggregated pageviews for specific CDS record
+            elif stat == 'pageviews':
+                key_type = 'date'
+                query = {
+                    "query": {
+                        "filtered": {
+                            "query": {
+                                "bool": {
+                                    "must": [
+                                        {
+                                            "match": {
+                                                "id_bibrec": pid.pid_value
+                                            }
+                                        },
+                                        {
+                                            "match": {
+                                                "_type":
+                                                    "events.cds_videos_pageviews"
+                                            }
                                         }
+                                    ]
+                                }
+                            },
+                            "filter": {
+                                "range": {
+                                    "@timestamp": {
+                                        "from": 0,
+                                        "to": "now"
                                     }
-                                ]
-                            }
-                        },
-                        "filter": {
-                            "range": {
-                                "@timestamp": {
-                                    "from": 0,
-                                    "to": "now"
                                 }
                             }
                         }
-                    }
-                },
-                "aggregations": {
-                    "by_time": {
-                        "date_histogram": {
-                            "field": "@timestamp",
-                            "interval": "day"
+                    },
+                    "aggregations": {
+                        "by_time": {
+                            "date_histogram": {
+                                "field": "@timestamp",
+                                "interval": "day"
+                            }
                         }
                     }
                 }
-            }
-            results = self.transform(
-                es.search(index=ES_INDEX, body=query),
-                stat,
-                key_type)
-
-        return make_response(jsonify(results), 200)
+                results = self.transform(
+                    es.search(index=ES_INDEX, body=query),
+                    stat,
+                    key_type)
+
+            return make_response(jsonify(results), 200)
+        except Exception:
+            return make_response("Couldn't connect to ES cluster.", 400)
```

Comment:
Can you move the `try/except` around the `es.search` call and change the exception instead?
I would also change the message because it might be returned, let's don't show such msg...
Why `400`?
```suggestion
            return make_response("Error getting statistics", 400)
```